src/main/java/io/cdap/plugin/cdc/transform/CDCStreamingToRecordTransform.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.cdc.transform;
+
+import com.google.cloud.Timestamp;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.Transform;
+import org.apache.spark.streaming.dstream.DStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ *
+ */
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name(CDCStreamingToRecordTransform.NAME)
+@Description("Transforms CDC record to normal record")
```

Comment:
I don't understand what this transform is being used for. What is a 'normal' record?

Reply:
this is used to transform a byte array to a ddl record that can be directly saved in sink

Reply:
I see. Does that mean the source is not usable unless it is connected to this transform? It seems like it would be better to put this into the source then?

Reply:
Though I didn't see any changes to what the source is outputting, I guess this is an issue with the existing plugin.

Reply:
I think we were not able directly join the source with BigQuery/BigTable sink ... So we extracted the dml part from and created a StructuredRecord with fields and values ...@zhaber I think we should probably rename the description. The plugin adds additional fields (CHANGE_TRACKING_VERSION and CDC_OP_TYPE) in the output StructuredRecord.

Reply:
After reading through the code, I now understand what this is doing. This seems like it's required because of poor design in the source. It should have been representing the row as a StructuredRecord instead of a Map of objects. Changing it would probably break the source though, so will keep this for now.